### ISSUE TITLE
Unable to delete regex version

### ISSUE THREAD

**Issue by VenkatSrinivasan** [firasdib/Regex101#1065]
## Bug Description
Unauthorised users are able to create a new version of regex 
(or An option to delete mistakenly created regex versions does not appear)

### Reproduction steps

1. As an unauthorized user, (i.e. not signed in)
2. Create a new version of https://regex101.com/r/OGc1mQ/2 (or another regex) by using "update regex" menu item
3. Try deleting the newly created version. 

### Expected Outcome
Unauthorised users should be prevented from creating new versions of regex
(Or An option to delete mistakenly created regex versions should be made available)

**Comment by Doqnach** [comment]
I believe the idea is that a user owns a revision of a regex, and not the whole "branch" itself. For instance, we help people with regex a lot on the IRC Channel (Freenode, #regex) and what happens is that a person provides us with a regex, and we provide them fixes or different solutions that may be more efficient. Your suggestion would prevent such help to be provided, or any form of team effort on a regex.

At the very least it would have to be a choice that should be given, more or less an option for authenticated users to "lock" their regex (can't save new revisions but can fork it).

Personally... I don't really see the point. You create a regex, and the final revision for you is the one you maybe keep in your library or link to from some outside source (code comment, ticket comment, etc). Only people you share the link with can made revisions so it's a matter of kindly asking the people you give the link to, to fork it instead of saving new revisions to the same branch. If it's something you want to link to publicly then I can see some benefit maybe for a lock option but even then you link to a specific revision anyways.

**Comment by VenkatSrinivasan** [comment]
By this "what happens is that a person provides us with a regex, and we provide them fixes or different solutions that may be more efficient.", you mean a new version is created when you provide a fix?

It wasn't clear to me how versions are meant to be managed, sorry.
Forking makes total sense infact that's what I wanted. However, clicked "update regex" (the one above) by mistake.

I was experimenting and ended up creating a new version. I may have made a mistake in the new version which I felt should not mislead someone else. Hence why I wanted to delete the version.

"more or less an option for authenticated users to "lock" their regex (can't save new revisions but can fork it)." - agree, will be useful

**Comment by Doqnach** [comment]
There is of course a difference between the intention of the website and your own requirements, and those are absolutely free to differ ;-].

If you are okay with it, I would suggest to close this issue and you comment on #173 that also goes in to this "lock" feature, to raise awareness for it.

**Comment by Doqnach** [comment]
Scratch that... I actually just found #173 and also read through the comments and the feature is already there :-P.

I just didn't know it ;-]. Sorry for the earlier false information.

**Comment by firasdib** [comment]
I believe all bases have been covered here. If there are any more questions, feel free to ask. Closing for now.

**Comment by ryanjm** [comment]
As a non-signed in user, I'd like to delete the version I created (accidentally). There doesn't appear to be a way for me to undo that action which is really bad since I put sensitive information in there which I don't want public.

How can I delete it?

**Comment by Doqnach** [comment]
You can check the developer console in your browser to see if the website can still identify you as the original owner, and you can see the delete link there.

But most important is to simply store the delete link somewhere next time, after saving the regex. Or signing in.

PS: always be wary when posting sensitive information to the web...

**Comment by ryanjm** [comment]
Agreed. I shouldn't have posted that information. I was thinking that everything would stay local and didn't think about the potential of it saving data even though I wasn't logged in. I'll be sure to be more careful in the future.

Unfortunately the developer console just says "Unable to retrieve delete code for permalink fragment ...".

Where would the delete link show up? I looked for it and couldn't find it. I've saved a new version and still can't find it and I'm logged in this time.
